rust/gen/src/main.rs

```diff
@@ -0,0 +1,119 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::{BTreeMap, BTreeSet};
+use std::fs::File;
+use std::io::Write;
+
+use anyhow::{anyhow, Context, Result};
+use serde::Deserialize;
+
+fn main() -> Result<()> {
+    let configs: BTreeMap<String, Config> =
+        serde_json::from_reader(File::open("data/config.json")?)?;
+    let mut labels = BTreeMap::<String, Vec<Label>>::new();
+    for (short, config) in configs {
+        if !config.in_scope_for_output_content_type {
+            continue;
+        }
+        if ["directory", "empty", "symlink"].contains(&short.as_str()) {
+            continue;
+        }
+        let target = match config.target_label {
+            Some(x) => x,
+            None => continue,
+        };
+        let exts = config.extensions;
+        let mime = unwrap(config.mime_type, "mime_type", &short)?;
+        let group = unwrap(config.group, "group", &short)?;
+        let magic = unwrap(config.magic, "magic", &short)?;
+        let long = unwrap(config.description, "description", &short)?;
+        let text = config.tags.iter().any(|x| *x == "text");
+        let label = Label { short, long, magic, group, mime, exts, text };
+        labels.entry(target).or_default().push(label);
+    }
+    let labels = labels;
+    let mut output = File::create("../lib/src/label.rs")?;
```

Comment:
Could you add a comment explaining why dynamically generating this file is necessary (or add it to the README.md of this directory)?